test/e2e/e2e_tests_suite_test.go

```diff
@@ -0,0 +1,221 @@
+	utilsHelpers := utils.New(cs)
+
+	nodesList, err := cs.CoreV1Interface.Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list nodes: %w", err)
+	}
+
+	for _, node := range nodesList.Items {
+		nodeRolde, err := types.GetNodeRole(&node)
+		if err != nil {
+			return err
+		}
+		if nodeRolde != role {
+			continue
+		}
+		ns := "commatrix-firewall-" + role
+		err = utilsHelpers.CreateNamespace(ns)
+		if err != nil {
+			return err
+		}
+		var cleanupErr error
+		defer func() {
+			if err := utilsHelpers.DeleteNamespace(ns); err != nil {
+				cleanupErr = err
+			}
+		}()
+
+		if cleanupErr != nil {
+			// Handle the error (log it, fail the test, etc.)
+			return cleanupErr
+		}
+		debugPod, err := utilsHelpers.CreatePodOnNode(node.Name, ns, consts.DefaultDebugPodImage)
+		if err != nil {
+			return fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+		}
+
+		defer func() {
+			err := utilsHelpers.DeletePod(debugPod)
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := []string{
+			"bash", "-c",
+			fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, installCmd)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		command = []string{
+			"bash", "-c", "/usr/sbin/nft -f nft/firewall.nft",
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, node.Name, err)
+		}
+
+		// Check the output of nft list ruleset
+		command = []string{"bash", "-c", "/usr/sbin/nft list ruleset"}
+		output, err := utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to list NFT ruleset on node %s: %w", node.Name, err)
+		}
+		fmt.Printf("Command output: %s\n", string(output))
+
+		if len(output) == 0 {
+			return fmt.Errorf("no output from 'nft list ruleset' on node %s: ", node.Name)
+		}
+		command = []string{
+			"bash", "-c",
+			"/usr/sbin/nft list ruleset > /etc/nftables.conf",
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", node.Name, err)
+		}
+
+		command = []string{"bash", "-c",
+			"systemctl enable nftables",
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to enable nftables %q on node %s: %w", command, node.Name, err)
+		}
+
+		artifactsDir, _ := os.LookupEnv("ARTIFACT_DIR")
```

Comment:
Should check the returned bool

Reply:
i did that like that becouse of local test i will add a print if not set but on the artifacts that should set